libs/core/execution_base/include/hpx/execution_base/traits/coroutine_traits.hpp

```diff
@@ -0,0 +1,231 @@
+        {
+        };
+
+    }    // namespace detail
+
+    // An Awaiter type is a type that implements the three special methods
+    // that are called as part of a co_await expression: await_ready,
+    // await_suspend and await_resume.
+
+    // https://www.open-std.org/jtc1/sc22/wg21/docs/papers/2018/p1288r0.pdf
+    // Lewis Baker - The rationale here is that for an awaiter object
+    // to be able to support being awaited in an arbitrary natural coroutine
+    // context it will generally need to type-erase the
+    // coroutine_handle<Promise> to coroutine_handle<void> so that it can
+    // store the continuation for an arbitrary coroutine-type.
+    // If theawait_suspend() method overload-set only has overloads that
+    // accept specific types of coroutine_handle<P> then it is only awaitable
+    // within specific contexts and thus we don’t consider it to satisfy the
+    // Awaiter concept.
+
+    template <typename Awaiter, typename Promise = void>
+    struct is_awaiter
+      : detail::is_awaiter_impl<detail::has_await_ready<Awaiter> &&
+                detail::has_await_resume<Awaiter> &&
+                (detail::has_await_suspend<Awaiter> ||
+                    detail::has_await_suspend_coro_handle<Awaiter, Promise>),
+            Awaiter, Promise>
+    {
+    };
+
+    template <typename Awaiter, typename Promise = void>
+    bool inline constexpr is_awaiter_v = is_awaiter<Awaiter, Promise>::value;
+
+    namespace detail {
+
+        template <typename Awaitable, typename = void>
+        constexpr bool has_member_operator_co_await_v = false;
+
+        template <typename Awaitable>
+        constexpr bool has_member_operator_co_await_v<Awaitable,
+            std::void_t<decltype(
+                std::declval<Awaitable>().operator co_await())>> = true;
+
+        template <typename Awaitable, typename = void>
+        constexpr bool has_free_operator_co_await_v = false;
+
+        template <typename Awaitable>
+        constexpr bool has_free_operator_co_await_v<Awaitable,
+            std::void_t<decltype(operator co_await(
+                std::declval<Awaitable>()))>> = true;
+
+        template <typename, typename = void, typename...>
+        constexpr bool has_await_transform{};
+
+        template <typename Promise, typename... Ts>
+        constexpr bool has_await_transform<Promise,
+            std::void_t<decltype(std::declval<Promise>().await_transform(
+                ((Ts &&) std::declval<Ts>())...))>,
+            Ts...> = true;
+
+    }    // namespace detail
+
+    // Returns the result of applying operator co_await() to the function’s
+    // argument, if the operator is defined, otherwise returns a reference
+    // to the input argument.
+    template <typename Awaitable>
+    decltype(auto) get_awaiter(Awaitable&& await, void*)
+    {
+        if constexpr (detail::has_member_operator_co_await_v<Awaitable>)
+        {
+            return ((Awaitable &&) await).operator co_await();
```

Comment:
Could you use `HPX_FORWARD(Awaitable, await).operator co_await()` instead, please? I know, it's the same ... just for consistency's sake again.

Reply:
Yes, definitely, I have seen `HPX_FORWARD` but was not sure if it did the same. Great!